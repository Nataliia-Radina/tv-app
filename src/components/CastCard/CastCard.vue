<template>
  <v-col
    :key="person.id"
    cols="6"
    sm="4"
    md="3"
    xl="2"
  >
    <v-card tile>
      <v-img :src="getImageSrc(person.person.image)" height="200px" :alt="person.name"></v-img>
      <v-card-text>
        <b>{{ person.person.name }}</b> as {{ person.character.name }}
      </v-card-text>
    </v-card>
  </v-col>
</template>

<script>
import ImageSrc from '@/mixins/ImageSrc'
export default {
  name: 'CardCast',
  mixins: [ImageSrc],
  props: {
    person: {
      type: Object,
      required: true
    }
  }
}
</script>
