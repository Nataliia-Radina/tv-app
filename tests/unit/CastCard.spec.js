import CastCard from '@/components/CastCard/CastCard.vue'
import { shallowMount } from '@vue/test-utils'
import Vue from 'vue'
import Vuetify from 'vuetify'
Vue.use(Vuetify)

it('expect to render CastCard component', () => {
  const person = {
    name: "Emilia Clarke",
    id: 1,
    person: {
      name: "Emilia Clarke",
      image: { medium: "src-medium-1", original: "src-original-1" },
    },
    character: {
      name: "Daenerys Targaryen",
      image: { medium: "src-medium-2", original: "src-original-2" },
    }
  }
  const wrapper = shallowMount(CastCard, {
    propsData: { person }
  })
  expect(wrapper).toMatchSnapshot()
})
