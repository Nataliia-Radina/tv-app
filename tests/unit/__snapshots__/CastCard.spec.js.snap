// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render CastCard component 1`] = `
<v-col-stub cols="6" sm="4" md="3" xl="2" tag="div">
  <v-card-stub loaderheight="4" tag="div" tile="true">
    <v-img-stub height="200px" alt="Emilia Clarke" options="[object Object]" position="center center" src="src-original-1" transition="fade-transition"></v-img-stub>
    <v-card-text-stub><b>Emilia Clarke</b> as Daenerys Targaryen
    </v-card-text-stub>
  </v-card-stub>
</v-col-stub>
`;
